### 1) copy raw data 

```
cp /home/knedlo/projects/rrg-ben/for_martin/2024_cliv/* /home/knedlo/projects/rrg-ben/knedlo/2024_cliv/raw_data
```

### 2) FastQC

### 3) trimmomatic

go to raw_data directory and check if there is `TruSeq2_and_3-PE-2.fa`. This file includes adaptor sequences. Then script is executed frem the same directory.

trimmomatic script:

```
#!/bin/sh
#SBATCH --job-name=trimmomatic
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=1
#SBATCH --time=96:00:00
#SBATCH --mem=8gb
#SBATCH --output=trimmomatic.%J.out
#SBATCH --error=trimmomatic.%J.err
#SBATCH --account=rrg-ben

# run by passing a directory argument like this
# sbatch ./2020_trimmomatic.sh path_to_raw_data
# TruSeq2_and_3-PE-2.fa must be present in the directory, from which the trimmomatic script is executed

module load StdEnv/2023
module load trimmomatic/0.39
# R1_001.fastq.gz
#v=1
#  Always use for-loop, prefix glob, check if exists file.
for file in $1/*R1_001.fastq.gz; do         # Use ./* ... NEVER bare *
  if [ -e "$file" ] ; then   # Check whether file exists.
      java -jar $EBROOTTRIMMOMATIC/trimmomatic-0.39.jar PE ${file::-15}R1_001.fastq.gz ${file::-15}R2_001.fastq.gz ${file::-15}_trim_R1.fq.gz ${file::-15}_trim_single_R1.fq.gz ${file::-15}_trim_R2.fq.gz ${file::-15}_trim_single_R2.fq.gz ILLUMINACLIP:TruSeq2_and_3-PE-2.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
  fi
done
```

```
../trimmomatic/2024_trimmomatic.sh ./
```

`./` = path `/home/knedlo/projects/rrg-ben/knedlo/2024_cliv/raw_data`

### 4) map trimmed sequences to X. laevis genome (bwa alignment)

Trimmed data were generated in the `raw_data` directory. Each trimmed read had two R1 and two R2 files (four primers were used for sequencing?)

folder made for each R1 and R2 trimmed reads (two directories for each individual) in the `trimmomatic` directory: e.g., `mkdir fem_cliv_CAS260427_S3_L001/`, move the CAS260427_S3_L001__trim_R1.fq.gz and CAS260427_S3_L001__trim_R2.fq.gz files in the fem_cliv_CAS260427_S3_L001/ folder using the `mv` command. `mkdir fem_cliv_CAS260427_S3_L002` and move the CAS260427_S3_L002__trim_R1.fq.gz and CAS260427_S3_L002__trim_R2.fq.gz files to the directory etc. Then move all single reads to the `trimmomatic` directory.

go to the first directory `/home/knedlo/projects/rrg-ben/knedlo/2024_cliv/trimmomatic/fem_cliv_CAS260392_S4_L001` use the '2020_align_paired_fq_to_ref.sh' script for each individual separately

```
#!/bin/sh
#SBATCH --job-name=bwa_align
#SBATCH --nodes=4
#SBATCH --ntasks-per-node=4
#SBATCH --time=168:00:00
#SBATCH --mem=32gb
#SBATCH --output=bwa_align.%J.out
#SBATCH --error=bwa_align.%J.err
#SBATCH --account=rrg-ben

# run by passing an argument like this (in the directory with the files)
# sbatch 2020_align_paired_fq_to_ref.sh pathandname_of_ref path_to_paired_fq_filez
# sbatch 2020_align_paired_fq_to_ref.sh /home/ben/projects/rrg-ben/ben/2018_Austin_XB_genome/Austin_genome/Xbo.v1.fa.gz pathtofqfilez

module load bwa/0.7.17
# module load samtools/1.10
module load StdEnv/2023  gcc/12.3 samtools/1.20

for file in ${2}/*_trim_R2.fq.gz ; do         # Use ./* ... NEVER bare *    
    if [ -e "$file" ] ; then   # Check whether file exists.
        bwa mem ${1} ${file::-14}_trim_R1.fq.gz ${file::-14}_trim_R2.fq.gz -t 16 | samtools view -Shu - | samtools sort - -o ${file::-14}_sorted.bam
        samtools index ${file::-14}_sorted.bam
  fi
done
```

the script were executed by: 
```
sbatch /home/knedlo/projects/rrg-ben/knedlo/ben_scripts/2020_align_paired_fq_to_ref.sh /home/knedlo/projects/rrg-ben/knedlo/laevis_genome/2021_XL_v10_refgenome/XL_v10.1_concatenatedscaffolds.fa .
```

### 5) merge 2 bam files in 1

`mkdir bams_merged`

after bwa_alignment jobs are done, move trimed files from individual directories to the directory `trimmomatic` and move directories of each individual to the directory `bams_merged`

a script for merging 2 bwa alignments into one:

```
#!/bin/sh
#SBATCH --job-name=merge2
#SBATCH --nodes=4
#SBATCH --ntasks-per-node=4
#SBATCH --time=32:00:00
#SBATCH --mem=16gb
#SBATCH --output=merge2.%J.out
#SBATCH --error=merge2.%J.err
#SBATCH --account=rrg-ben

# run by passing an argument like this (in the directory with the files)
# sbatch 2024_samtools_merge2.sh merged_path_and_file in1_path_and_file in2_path_and_file 

module load StdEnv/2023  gcc/12.3 samtools/1.20

samtools merge ${1} ${2} ${3}

samtools index ${1}
```

go to the directory `bams_merged` and execute the script for `CAS262485`:
```
sbatch 2024_samtools_merge2.sh fem_cliv_CAS262485.merged.bam fem_cliv_CAS262485_S2_L001/CAS262485_S2_L001__sorted.bam fem_cliv_CAS262485_S2_L002/CAS262485_S2_L002__sorted.bam
```

### 6) read groups using picard

move all sorted directories with files (bwa aligned) to the directory `bwa_alignment` using `mv ../bams_merged/*L001 ../bams_merged/*L002 .`

`.` = `bwa_alignment`

make directory for each individual in `bams_merged` and move merged.bam files there using move command

read group script:

```
#!/bin/sh
#SBATCH --job-name=readgroups
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=1
#SBATCH --time=24:00:00
#SBATCH --mem=8gb
#SBATCH --output=readgroups.%J.out
#SBATCH --error=readgroups.%J.err
#SBATCH --account=def-ben

# run by passing an argument like this
# sbatch 2022_picard_add_read_groups_and_index.sh bamfile_prefix

# module load picard/2.23.3
module load picard/3.1.0

    java -jar $EBROOTPICARD/picard.jar AddOrReplaceReadGroups I=${1}.bam O=${1}_rg.bam RGID=4 RGLB=${1} RGPL=ILLUMINA RGPU=${1} RGSM=${1}

# module load StdEnv/2020 samtools/1.12
module load StdEnv/2023 samtools/1.20
samtools index ${1}_rg.bam
```

execute read-group script for each individual separately:
```
sbatch /home/knedlo/projects/rrg-ben/knedlo/ben_scripts/2022_picard_add_read_groups_and_index.sh fem_cliv_CAS260392_merged
```

### 7) call haplotype aka haplotype caller

`mkdir read_groups`

move merged_rg.bam and merged_rg.bam.bai files from `bams_merged` to `read_groups` directory

copy directories from `read_groups` to `haplotype_caller`. After copying directories with merged_rg.bam and merged_rg.bam.bai files, haplotype caller can be done for each directory separately

```
#!/bin/sh
#SBATCH --job-name=HaplotypeCaller
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=1
#SBATCH --time=96:00:00
#SBATCH --mem=30gb
#SBATCH --output=HaplotypeCaller.%J.out
#SBATCH --error=HaplotypeCaller.%J.err
#SBATCH --account=rrg-ben


# This script will read in the *_sorted.bam file names in a directory, and 
# make and execute the GATK command "RealignerTargetCreator" on these files. 

# execute like this:
# sbatch 2021_HaplotypeCaller.sh ref dir_of_bam chr
# sbatch 2021_HaplotypeCaller.sh /home/knedlo/projects/rrg-ben/knedlo/laevis_genome/2021_XL_v10_refgenome/XL_v10.1_concatenatedscaffolds.fa bamfiledirectory chr

# module load nixpkgs/16.09 gatk/4.1.0.0
module load gatk/4.4.0.0 StdEnv/2023

for file in ${2}*_rg.bam
do
    gatk --java-options -Xmx24G HaplotypeCaller  -I ${file} -R ${1} -L ${3} -O ${file}_${3}.g.vcf -ERC GVCF
done
```

```
for x in {1..8}; do sbatch /home/knedlo/projects/rrg-ben/knedlo/ben_scripts/2024_HaplotypeCaller.sh /home/knedlo/projects/rrg-ben/knedlo/laevis_genome/2021_XL_v10_refgenome/XL_v10.1_concatenatedscaffolds.fa ./ Chr$x\L; done
```

```
for x in {1..8}; do sbatch /home/knedlo/projects/rrg-ben/knedlo/ben_scripts/2024_HaplotypeCaller.sh /home/knedlo/projects/rrg-ben/knedlo/laevis_genome/2021_XL_v10_refgenome/XL_v10.1_concatenatedscaffolds.fa ./ Chr$x\S; done
```

```
sbatch /home/knedlo/projects/rrg-ben/knedlo/ben_scripts/2024_HaplotypeCaller.sh /home/knedlo/projects/rrg-ben/knedlo/laevis_genome/2021_XL_v10_refgenome/XL_v10.1_concatenatedscaffolds.fa ./ Chr9_10L
```
```
sbatch /home/knedlo/projects/rrg-ben/knedlo/ben_scripts/2024_HaplotypeCaller.sh /home/knedlo/projects/rrg-ben/knedlo/laevis_genome/2021_XL_v10_refgenome/XL_v10.1_concatenatedscaffolds.fa ./ Chr9_10S
```

```
sbatch /home/knedlo/projects/rrg-ben/knedlo/ben_scripts/2024_HaplotypeCaller.sh /home/knedlo/projects/rrg-ben/knedlo/laevis_genome/2021_XL_v10_refgenome/XL_v10.1_concatenatedscaffolds.fa ./ Scaffolds
```

`./` is the directory of each individual `/home/knedlo/projects/rrg-ben/knedlo/2024_cliv/haplotype_caller/fem_cliv_CAS260392` etc.

### 8) CombineGVCFs

all files generated by haplotype caller were moved from individual directories to one directory `07_haplotype_caller`

script for combine GVCF was executed form the the `07_haplotype_caller` directory

```
#!/bin/sh
#SBATCH --job-name=CombineGVCFs
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=1
#SBATCH --time=48:00:00
#SBATCH --mem=12gb
#SBATCH --output=CombineGVCFs.%J.out
#SBATCH --error=CombineGVCFs.%J.err
#SBATCH --account=def-ben

# for graham.computecanada change account def-ben to rrg-ben 
# This script will read in the *.g.vcf file names in a directory, and 
# make and execute the GATK command "GenotypeGVCFs" on these files. 

# execute like this:
# sbatch ../../ben_scripts/2021_CombineGVCFs.sh /home/knedlo/projects/rrg-ben/knedlo/laevis_genome/2021_XL_v10_refgenome/XL_v10.1_concatenatedscaffolds.fa ./ Chr1L

module load nixpkgs/16.09 gatk/4.1.0.0

commandline="gatk --java-options -Xmx10G CombineGVCFs -R ${1}"
for file in ${2}*${3}*g.vcf
do
    commandline+=" -V ${file}"
done

commandline+=" -L ${3} -O ${2}allsites_${3}.g.vcf.gz"

${commandline}
```

using commands:

```
for x in {1..8}; do sbatch /home/knedlo/projects/def-ben/knedlo/ben_scripts/2021_CombineGVCFs.sh /home/knedlo/projects/def-ben/knedlo/laevis_genome/2021_XL_v10_refgenome/XL_v10.1_concatenatedscaffolds.fa ./ Chr$x\L; done
```
```
for x in {1..8}; do sbatch /home/knedlo/projects/def-ben/knedlo/ben_scripts/2021_CombineGVCFs.sh /home/knedlo/projects/def-ben/knedlo/laevis_genome/2021_XL_v10_refgenome/XL_v10.1_concatenatedscaffolds.fa ./ Chr$x\S; done
```
```
sbatch /home/knedlo/projects/def-ben/knedlo/ben_scripts/2021_CombineGVCFs.sh /home/knedlo/projects/def-ben/knedlo/laevis_genome/2021_XL_v10_refgenome/XL_v10.1_concatenatedscaffolds.fa ./ Chr9_10L
```
```
sbatch /home/knedlo/projects/def-ben/knedlo/ben_scripts/2021_CombineGVCFs.sh /home/knedlo/projects/def-ben/knedlo/laevis_genome/2021_XL_v10_refgenome/XL_v10.1_concatenatedscaffolds.fa ./ Chr9_10S
```
```
sbatch /home/knedlo/projects/def-ben/knedlo/ben_scripts/2021_CombineGVCFs.sh /home/knedlo/projects/def-ben/knedlo/laevis_genome/2021_XL_v10_refgenome/XL_v10.1_concatenatedscaffolds.fa ./ Scaffolds
```
