### 1) copy raw data 

```
cp /home/knedlo/projects/rrg-ben/for_martin/2024_cliv/* /home/knedlo/projects/rrg-ben/knedlo/2024_cliv/raw_data
```

### 2) FastQC

### 3) trimmomatic

trimmomatic script:

```
#!/bin/sh
#SBATCH --job-name=trimmomatic
#SBATCH --nodes=1
#SBATCH --ntasks-per-node=1
#SBATCH --time=96:00:00
#SBATCH --mem=8gb
#SBATCH --output=trimmomatic.%J.out
#SBATCH --error=trimmomatic.%J.err
#SBATCH --account=rrg-ben

# run by passing a directory argument like this
# sbatch ./2020_trimmomatic.sh path_to_raw_data


module load StdEnv/2023
module load trimmomatic/0.39
# R1_001.fastq.gz
#v=1
#  Always use for-loop, prefix glob, check if exists file.
for file in $1/*R1_001.fastq.gz; do         # Use ./* ... NEVER bare *
  if [ -e "$file" ] ; then   # Check whether file exists.
      java -jar $EBROOTTRIMMOMATIC/trimmomatic-0.39.jar PE ${file::-15}R1_001.fastq.gz ${file::-15}R2_001.fastq.gz ${file::-15}_trim_R1.fq.gz ${file::-15}_trim_single_R1.fq.gz ${file::-15}_trim_R2.fq.gz ${file::-15}_trim_single_R2.fq.gz ILLUMINACLIP:TruSeq2_and_3-PE-2.fa:2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:36
  fi
done
```

```
sbatch /home/knedlo/projects/rrg-ben/knedlo/2024_cliv/trimmomatic/2024_trimmomatic.sh ../raw_data/
```

`../raw_data/` = absolute path `/home/knedlo/projects/rrg-ben/knedlo/2024_cliv/raw_data`, i.e., data were generated in the raw data directory regardless the current position

### 4) map trimmed sequences to X. laevis genome (bwa alignment)

each trimmed read had two R1 and two R2 files (four primers were used for sequencing?)

folder made for each R1 and R2 trimmed reads (two directories for each individual) : e.g., mkdir fem_larg_BJE1581/, move the fem_larg_BJE1581_S8_trim_R1.fq.gz and fem_larg_BJE1581_S8_trim_R2.fq.gz files in the fem_larg_BJE1581 folder using mv command

mkdir fem_larg_BJE1582/ and move the fem_larg_BJE1582_S9_trim_R1.fq.gz and fem_larg_BJE1582_S9_trim_R2.fq.gz files there etc.

use the '2024_align_paired_fq_to_ref.sh' script ('martin_csripts' folder because modules in the script were updated) for each individual separately



### 5) merge 2 bam files in 1

mkdir bams_combined

go to the ben_scripts folder and check if there are 3 files with the name AMNH17292: ls ../raw_data/*/*AMNH17292* shows 12 files: 3 R1_trimmed, 3 R2_trimmed, 3 sorted.bam, and 3 sorted.bam.bai ls ../raw_data/*/*AMNH17292*bam* shows 6 files: 3 sorted.bam, and 3 sorted.bam.bai ls ../raw_data/*/*AMNH17292*bam shows 3 sorted.bam files /*/ involves all folders in the raw_data folder (all temp folders)

a script for merging:

```
#!/bin/sh
#SBATCH --job-name=merge2
#SBATCH --nodes=4
#SBATCH --ntasks-per-node=4
#SBATCH --time=32:00:00
#SBATCH --mem=16gb
#SBATCH --output=merge2.%J.out
#SBATCH --error=merge2.%J.err
#SBATCH --account=rrg-ben

# run by passing an argument like this (in the directory with the files)
# sbatch 2024_samtools_merge2.sh merged_path_and_file in1_path_and_file in2_path_and_file 

module load StdEnv/2023  gcc/12.3 samtools/1.20

samtools merge ${1} ${2} ${3}

samtools index ${1}
```
